### ISSUE TITLE
Capture traffic with PING frames 

### ISSUE THREAD

**Issue by luisespla** [nghttp2/nghttp2#345]
Hello,
I would like to capture traffic with PING frames, when I've been trying it, I could see PING frames in nghttp2 server log but then, PING frames aren't in the capture. Does anyone have idea why I can't it or How can I do it?
BR

**Comment by lespla** [comment]
Can anybody help me?
BR

**Comment by tatsuhiro-t** [comment]
If either endpoint gets PING frame received, there must be a packet containing it.
Are you using wireshark?  One packet may contain several HTTP/2 frames.  PING is small, it may be buried somewhere.
cc: @alagoutte

**Comment by lespla** [comment]
Yes, I'm using wireshark and tcpdump for capturing the traffic but it doesn't work with ping frames. Then capturing I look for it with http2.type==6. 
Tomorrow, I'm trying it again. 
BR

**Comment by alagoutte** [comment]
What Wireshark release ? can you attach a pcap sample ?

**Comment by lespla** [comment]
I'm using tcpdum version 3.9.8 and nghttp with proxy, server and client 1.3.0.
 I've attached capture file, I've changed extension file from pcap to png.
Thanks
![ping](https://cloud.githubusercontent.com/assets/4354214/9844055/69d4773e-5ac0-11e5-8e6f-db105c938068.png)

**Comment by lespla** [comment]
Hello again,
Is it available you attach a h2c capture file with ping frame?
Best regards

**Comment by alagoutte** [comment]
After a quick look, there is some issue with resseamble HTTP2 (and HTTP Upgrade...)

Can you create a issue on Wireshark bugtracker with your pcap ? https://bugs.wireshark.org

**Comment by lespla** [comment]
I wrote in issue a wireshark about upgrade 45 days ago. I'm going to try it without upgrade but I have done a http2 dissector with scapy and I don't see ping frame in that capture file either. I don't already use wireshark for nothing.
Do you have h2c capture file with ping frame?
Thanks a lot

**Comment by alagoutte** [comment]
What the bug id ?

**Comment by lespla** [comment]
it had problems with upgrade and without upgrade
https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=11331

**Comment by lespla** [comment]
I would thank you if you send me h2c capture file with ping frame.
BR

**Comment by alagoutte** [comment]
can you open a new issue with your pcap ? because there is a another issue (about resseamble i think)

i have no h2c pcap with ping...

**Comment by lespla** [comment]
Hello, 
Are you talking about wireshark issue?  I think it's not wireshark problem, because I used tcpdump for capturing and ping frame it's not in capture file. I used wireshark and another dissector done for me and I can't see ping frame. I think is nghttp2 problem. It shows it in its logs but it doesn't send it. 
Best regards

**Comment by alagoutte** [comment]
There is no HTTP2 dissector on tcpdump...

**Comment by lespla** [comment]
I've done it with scapy in Python not in tcpdump. With tcpdump only capture
and then analyzing with scapy dissector.

**Comment by alagoutte** [comment]
@lespla Ok !

But i will great to open a another issue because wrong decoding actually...

**Comment by tatsuhiro-t** [comment]
nghttp client and nghttpd server actually send/receive PING frames.  They must be on wire.

**Comment by lespla** [comment]
Do you have a h2c capture file with ping frame?
BR

**Comment by alagoutte** [comment]
it is confirmed there is a issue with Wireshark and HTTP Upgrade https://code.wireshark.org/review/#/c/10541/

And no check  about Data resseamble

**Comment by lespla** [comment]
But it's no related with PING frames :-)

**Comment by alagoutte** [comment]
Yes...
There is a another issue (thanks for your pcap) but Yes, there is no PING frames.

**Comment by lespla** [comment]
What problem is there? Could you give me more details?
BR

**Comment by alagoutte** [comment]
https://code.wireshark.org/review/#/c/10544/
There is a issue on Wireshark code about desegmentation when using h2c
but it is no fixed on master :-)

**Comment by alagoutte** [comment]
For information, if we always need PING Frame, you can try to connect to twitter (using SSLKEYFILELOG) and we will say some PING frame
